cellfinder_napari/detect_utils.py

```diff
@@ -0,0 +1,186 @@
+from dataclasses import dataclass, asdict
+from pathlib import Path
+
+import napari
+from cellfinder_core.main import main as cellfinder_run
+from imlib.cells.cells import Cell
+from napari.qt.threading import thread_worker
+
+from cellfinder_napari.utils import cells_to_array
+
+
+@dataclass
+class InputContainer:
+    """Base data class to group inputs into a container, with persistent defaults"""
+
+    persistent_defaults = (
+        {}
+    )  # some elements need to be remembered without having to create a new instance, for reset button
+
+    @classmethod
+    def numerical_widget(
+        cls, key, custom_label=None, step=0.1, min=None, max=None
```

Comment:
I think we should really try to include typing as much as possible. Can be a lifesaver.

We could also add `mypy` to the `pre-commit` hook. What do you think?

Reply:
100%. I'm (slowly) introducing it to other BG tools.

Reply:
I'm going to raise a PR including this soon